packages/authentication-client/src/core.ts

```diff
@@ -152,7 +152,9 @@ export class AuthenticationClient {
     // For NotAuthenticated, PaymentError, Forbidden, NotFound, MethodNotAllowed, NotAcceptable
     // errors, remove the access token
     if (error.code > 400 && error.code < 408) {
-      const promise = this.removeAccessToken().then(() => this.reset())
+      const promise = (this.storage as StorageWrapper).storage
```

Comment:
My vote would be to leave this line alone, and let it make the request to `removeAccessToken`.  Then update line 136 in `removeAccessToken` to read

```ts
    return this.storage?.removeItem(this.options.storageKey)
```

In other words, let's let `removeAccessToken` check for `storage` instead of the error handler.